src/connectionController.ts

```diff
@@ -989,9 +982,13 @@ export default class ConnectionController {
     this._connectingConnectionId = '';
   }
 
-  // Exposed for testing.
   setActiveDataService(newDataService: DataService): void {
     this._activeDataService = newDataService;
+
+    // Disconnect the extension if the MongoDB client is closed.
+    this._activeDataService?.once('close', () => {
+      void this.disconnect();
```

Comment:
When does the `close` event occur? Doesn't it also happen when we disconnect?
Should we cleanup these listeners? Or are they already cleaned up here https://github.com/mongodb-js/compass/blob/e7b5391b6b80f5524378da2e1d3988f7da61715b/packages/data-service/src/data-service.ts#L2634

Reply:
Correct, when data service disconnects, the `_cleanup()` helper removes all listeners.